scheduler/src/cook/mesos/mesos_compute_cluster.clj

```diff
@@ -65,17 +259,18 @@
 (defn get-mesos-compute-cluster
   "Process one mesos cluster specification, returning the entity id of the corresponding compute-cluster,
   creating the cluster if it does not exist. Warning: Not idempotent. Only call once "
-  [conn {:keys [compute-cluster-name framework-id] :as mesos-cluster}]
-  {:pre [compute-cluster-name
-         framework-id]}
-  (let [cluster-entity-id (get-mesos-cluster-entity-id (d/db conn) mesos-cluster)]
-    (when-not cluster-entity-id
-      (cc/write-compute-cluster conn (mesos-cluster->compute-cluster-map-for-datomic mesos-cluster)))
-    (->MesosComputeCluster
-      compute-cluster-name
-      framework-id
-      (or cluster-entity-id (get-mesos-cluster-entity-id (d/db conn) mesos-cluster))
-      (atom nil))))
+  ([conn create-mesos-compute-cluster mesos-cluster]
+    (get-mesos-compute-cluster conn create-mesos-compute-cluster mesos-cluster nil))
+  ([conn create-mesos-compute-cluster {:keys [compute-cluster-name framework-id] :as mesos-cluster} driver]
+   {:pre [compute-cluster-name
+          framework-id]}
+   (let [cluster-entity-id (get-mesos-cluster-entity-id (d/db conn) mesos-cluster)]
+     (when-not cluster-entity-id
+       (cc/write-compute-cluster conn (mesos-cluster->compute-cluster-map-for-datomic mesos-cluster)))
+     (create-mesos-compute-cluster compute-cluster-name
```

Comment:
Still don't like. :(  But the least worst evil.

Reply:
Yeah, hopefully some of this can be further cleaned up later. But I wanted to stick to moving things as much as possible in this PR and avoid changing too much.